kernel/src/syscall/execve.rs

```diff
@@ -141,12 +142,11 @@ fn do_execve(
     // set cpu context to default
     *user_context.general_regs_mut() = RawGeneralRegs::default();
     user_context.set_tls_pointer(0);
-    *user_context.fpu_state_mut() = FpuState::default();
-    // FIXME: how to reset the FPU state correctly? Before returning to the user space,
-    // the kernel will call `handle_pending_signal`, which may update the CPU states so that
-    // when the kernel switches to the user mode, the control of the CPU will be handed over
-    // to the user-registered signal handlers.
-    user_context.fpu_state().restore();
+    // reset the CPU's FPU state
+    if let Some(fpu_state) = Task::current().unwrap().fpu_state() {
+        *fpu_state.borrow_mut() = FpuState::default();
+        fpu_state.borrow().restore();
```

Comment:
`RefCell::borrow` is not zero-cost, it also incurs runtime cost. So we should avoid it if we can.

```rust
let fpu_state = fpu_state.borrow_mut();
*fpu_state = = FpuState::default();
fpu_state.restore()
```